developer/src/kmc-kmn/src/kmw-compiler/constants.ts

```diff
@@ -21,9 +21,27 @@ export const
     ['*ShiftedLock*', '*Shifted*'],
     ['*ZWNJ*', '<|>'],
     ['*ZWNJiOS*', '<|>'],
-    ['*ZWNJAndroid*', '<|>']
+    ['*ZWNJAndroid*', '<|>'],
+  ],
+  // these names were added in Keyman 17
+  CSpecialText17: string = '*Sp*\0*NBSp*\0*NarNBSp*\0*EnQ*\0*EmQ*\0*EnSp*\0*EmSp*\0*PunctSp*\0' +
+    '*ThSp*\0*HSp*\0*ZWSp*\0*ZWJ*\0*WJ*\0*CGJ*\0*LTRM*\0*RTLM*\0*SH*\0*HTab*\0',
+  CSpecialText17ZWNJ: string = '*ZWNJGeneric*',
+  CSpecialText17Map: string[][] = [
+    ['*ZWNJGeneric*', '<|>']
   ];
 
+  // Map for checking minimum versions and Special Text
+  export const CSpecialText = new Map<number, string>([
+    [10, CSpecialText10],
+    [11, CSpecialText10],
+    [12, CSpecialText10],
+    [13, CSpecialText10],
+    [14, CSpecialText14 + CSpecialText10],
+    [15, CSpecialText14 + CSpecialText10],
+    [16, CSpecialText14 + CSpecialText10],
+    [17, CSpecialText17 + CSpecialText14 + CSpecialText10]
+  ]);
```

Comment:
```suggestion
  export const CSpecialText = new Map<number, string>([
    [KMX.KMXFile.VERSION_100, CSpecialText10],
    [KMX.KMXFile.VERSION_110, CSpecialText10],
    [KMX.KMXFile.VERSION_120, CSpecialText10],
    [KMX.KMXFile.VERSION_130, CSpecialText10],
    [KMX.KMXFile.VERSION_140, CSpecialText14 + CSpecialText10],
    [KMX.KMXFile.VERSION_150, CSpecialText14 + CSpecialText10],
    [KMX.KMXFile.VERSION_160, CSpecialText14 + CSpecialText10],
    [KMX.KMXFile.VERSION_170, CSpecialText17 + CSpecialText14 + CSpecialText10]
  ]);
  export const CSpecialTextMinVer = KMX.KMXFile.VERSION_100;
  export const CSpecialTextMaxVer = KMX.KMXFile.VERSION_170;
```

Reply:
ah, this involved updating KMX to add `VERSION_110`, `VERSION_120`, `VERSION_130`